src/logcollector/read_audit.c

```diff
@@ -48,22 +48,34 @@ void *read_audit(logreader *lf, int *rc, int drop_it) {
     char *id;
     char *p;
     size_t z;
-    long offset;
-    long rbytes;
+    int64_t offset;
+    int64_t rbytes;
+
     int lines = 0;
 
     *rc = 0;
 
     for (offset = w_ftell(lf->fp); fgets(buffer, OS_MAXSTR, lf->fp) && (!maximum_lines || lines < maximum_lines); offset += rbytes) {
         rbytes = w_ftell(lf->fp) - offset;
+
+        /* Flow control */
+        if ( rbytes <= 0) {
+            break;
+        }
+
         lines++;
 
         if (buffer[rbytes - 1] == '\n') {
             buffer[rbytes - 1] = '\0';
 
-            if ((long)strlen(buffer) != rbytes - 1)
+            if ((int64_t)strlen(buffer) != rbytes - 1)
             {
+                #ifdef WIN32
+                mdebug2("Line in '%s' contains some zero-bytes (valid=%lld / total=%lld). Dropping line.", lf->file, (int64_t)strlen(buffer), rbytes - 1);
```

Comment:
Can we simplify this bunch of repeated messages?
For instance:
```c
#ifdef WIN32
#define FTELL_TT "lld"
#else
#define FTELL_TT "ld"
#endif
// (...)
mdebug2("Line in '%s' contains some zero-bytes (valid=%" FTELL_TT " / total=%" FTELL_TT "). Dropping line.", lf->file, strlen(buffer), rbytes - 1);
```

Same fix if `strlen(buffer)` produces a warning.